pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -383,6 +383,19 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
     }
   }
 
+  public RebalanceResult.Status rebalanceStatus(String tableNameWithType, IdealState idealState,
+      ExternalView externalView) {
+
+    if (idealState == null) {
+      throw new IllegalStateException("ideal state is empty");
+    }
+    if (externalView != null && isExternalViewConverged(tableNameWithType, externalView.getRecord().getMapFields(),
```

Comment:
Do we want to consider the situation where external view never converges with what ongoing rebalancer expects or external view ends up having segments in ERROR state ? 

Looks like we will continue returning IN_PROGRESS in such scenarios and that could be misleading ? Probably also depends on what is the call to action for the user/admin who uses this API...

Reply:
HI @siddharthteotia, sorry I updated the above code before reading this comment.Currently, the external view returns  false if it is not converged yet and logs an error if the external view ends up having segments in error .I believe it is best to return the status FAILED instead of logging. Also, would it be ok to return NO_OP if the external view is still null when the api is called .cc @Jackie-Jiang